model/utils/data_preprocess.py

```diff
@@ -62,8 +61,10 @@ def encode_onehot(df: pd.DataFrame) -> pd.DataFrame:
     return df
 
 
-def preprocess_data(df: pd.DataFrame, df_test: Optional[pd.DataFrame] = None) -> pd.DataFrame:
-    uncorrelated_col, insignificant_col, missing_values_col = get_data_for_preprocessing(df, treshhold=0.05)
+def preprocess_data(mlflow: MLFlowHandler, df: pd.DataFrame, df_test: Optional[pd.DataFrame] = None) -> pd.DataFrame:
+    uncorrelated_col, insignificant_col, missing_values_col = get_data_for_preprocessing(
+        df, treshhold=0.05, mlflow=mlflow
+    )
```

Comment:
This function is also used in main function that logs preprocessed data to MLFlow for every iteration.
Preprocess data calls the get_data_for_preprocessing(this logs the values used for preprocessing). After preprocessing the preprocessed data is logged to mlflow. 
Then I would need more information on how to rewrite this. This data shouldn't be logged?

Reply:
Can't you run it once before training and then train the model on the preprocessed data? It's good practice as you can then cache those preprocessed values and speed up the training.

What I was suggesting before is to remove the first argument `mlflow: MLFlowHandler` from the function call and then just directly use the API of this object to upload the results of this function. This is the separation of concerns. Right now this function does 2 things - preprocessing the data AND uploading them to the MLFlow. This might have been 2 functions.